### ISSUE TITLE
Support waiting on multiple conditions in kubectl wait

### ISSUE THREAD

**Issue by aleks-m** [kubernetes/kubernetes#100248]
Hi.

I am using `kubectl wait` to monitor long-running jobs from GitLab CI. Currently I can wait only on one condition, e.g.:
```
kubectl wait --for=condition=complete --timeout=$KUBECTL_JOB_WAIT_TIMEOUT job/$job_id --namespace=$GKE_NAMESPACE
```
In case of a job failure (especially with `DeadlineExceeded` or `CreateContainerConfigError `) - `kubectl` keeps waiting for a kubectl timeout, in my case it is 40 minutes.

It would be nice to be able to wait on multiple conditions, e.g.:
```
kubectl wait --for=condition=complete,failed --timeout=$KUBECTL_JOB_WAIT_TIMEOUT job/$job_id --namespace=$GKE_NAMESPACE
```

That would allow kubectl to return immediately after a job failure and save tons of my time.

**Comment by rikatz** [comment]
This is a duplicate of https://github.com/kubernetes/kubernetes/issues/95759?

**Comment by caleb15** [comment]
@rikatz This issue is for OR'ing of conditions, whereas the other one is for AND'ing. Whether or not this issue is a duplicate depends on whether the implementer wants to tackle both features at once because they are related or do them separately.

**Comment by caleb15** [comment]
It looks like https://github.com/kubernetes/kubernetes/pull/97411 is doing both of the features, so one of these issues should be closed as a duplicate and the other reopened.
